src/types/geolocationTypes.ts

```diff
@@ -1 +1,25 @@
+import { parseJson } from '@/utils/jsonUtil';
+
 export type LatLon = Pick<GeolocationCoordinates, 'latitude' | 'longitude'>;
+
+export function isLatLon(value: unknown): value is LatLon {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    'latitude' in value &&
+    'longitude' in value
```

Comment:
maybe check for numeric values?
```suggestion
    'longitude' in value &&
    typeof value.latitude === 'number' &&
    typeof value.longitude === 'number'
```